Cast Exception Message in Integration Tests

Seeing below message in the logs. 

`[INFO] Running com.yahoo.elide.async.integration.tests.AsyncIT
15:03:02.408 [pool-8-thread-1] ERROR c.y.e.a.s.dao.DefaultAsyncAPIDAO.loadAsyncAPICollection(DefaultAsyncAPIDAO.java:199) - Exception: {}
java.lang.ClassCastException: com.yahoo.elide.datastores.hibernate5.ScrollableIterator cannot be cast to java.util.Collection
    at com.yahoo.elide.async.service.dao.DefaultAsyncAPIDAO.loadAsyncAPICollection(DefaultAsyncAPIDAO.java:189)
    at com.yahoo.elide.async.service.thread.AsyncAPICancelThread.cancelAsyncAPI(AsyncAPICancelThread.java:81)
    at com.yahoo.elide.async.service.thread.AsyncAPICancelThread.run(AsyncAPICancelThread.java:54)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
15:03:02.411 [pool-8-thread-1] ERROR c.y.e.a.s.t.AsyncAPICancelThread.cancelAsyncAPI(AsyncAPICancelThread.java:127) - Exception in scheduled cancellation: {}
java.lang.IllegalStateException: java.lang.ClassCastException: com.yahoo.elide.datastores.hibernate5.ScrollableIterator cannot be cast to java.util.Collection
    at com.yahoo.elide.async.service.dao.DefaultAsyncAPIDAO.loadAsyncAPICollection(DefaultAsyncAPIDAO.java:200)
    at com.yahoo.elide.async.service.thread.AsyncAPICancelThread.cancelAsyncAPI(AsyncAPICancelThread.java:81)
    at com.yahoo.elide.async.service.thread.AsyncAPICancelThread.run(AsyncAPICancelThread.java:54)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassCastException: com.yahoo.elide.datastores.hibernate5.ScrollableIterator cannot be cast to java.util.Collection
    at com.yahoo.elide.async.service.dao.DefaultAsyncAPIDAO.loadAsyncAPICollection(DefaultAsyncAPIDAO.java:189)
    ... 9 common frames omitted`


## Expected Behavior
The exception should not be happening.

## Current Behavior
The exception is happening.

## Possible Solution
DefaultAsyncAPIDAO should work with Iterables instead of collections.

--- a/elide-async/src/main/java/com/yahoo/elide/async/service/dao/AsyncAPIDAO.java
+++ b/elide-async/src/main/java/com/yahoo/elide/async/service/dao/AsyncAPIDAO.java
@@ -10,8 +10,6 @@ import com.yahoo.elide.async.models.AsyncAPIResult;
 import com.yahoo.elide.async.models.QueryStatus;
 import com.yahoo.elide.core.filter.expression.FilterExpression;
 
-import java.util.Collection;
-
 /**
  * Utility interface which uses the elide datastore to modify and create AsyncAPI and AsyncAPIResult Objects.
  */
@@ -37,32 +35,32 @@ public interface AsyncAPIDAO {
             String asyncAPIId, Class<T> type);
 
     /**
-     * This method deletes a collection of AsyncAPI and its associated AsyncAPIResult objects from database and
-     * returns the objects deleted.
+     * This method deletes a Iterable of AsyncAPI and its associated AsyncAPIResult objects from database
+     * based on a filter expression, and returns the objects deleted.
      * @param filterExpression filter expression to delete AsyncAPI Objects based on
      * @param type AsyncAPI Type Implementation.
-     * @return query object list deleted
+     * @return query object Iterable deleted
      */
-    public <T extends AsyncAPI> Collection<T> deleteAsyncAPIAndResultCollection(
+    public <T extends AsyncAPI> Iterable<T> deleteAsyncAPIAndResultByFilter(
             FilterExpression filterExpression, Class<T> type);
 
     /**
-     * This method updates the status for a collection of AsyncAPI objects from database and
+     * This method updates the status for a Iterable of AsyncAPI objects from database based on a filter expression, and
      * returns the objects updated.
      * @param  filterExpression filter expression to update AsyncAPI Objects based on
      * @param status status to be updated
      * @param type AsyncAPI Type Implementation.
-     * @return query object list updated
+     * @return query object Iterable updated
      */
-    public <T extends AsyncAPI> Collection<T> updateStatusAsyncAPICollection(FilterExpression filterExpression,
+    public <T extends AsyncAPI> Iterable<T> updateStatusAsyncAPIByFilter(FilterExpression filterExpression,
             QueryStatus status, Class<T> type);
     /**
-     * This method gets a collection of AsyncAPI objects from database and
+     * This method gets a Iterable of AsyncAPI objects from database and
      * returns the objects.
      * @param filterExpression filter expression to cancel AsyncAPI Objects based on
      * @param type AsyncAPI Type Implementation.
-     * @return query object list updated
+     * @return query object Iterable loaded
      */
-    public <T extends AsyncAPI> Collection<T> loadAsyncAPICollection(FilterExpression filterExpression,
+    public <T extends AsyncAPI> Iterable<T> loadAsyncAPIByFilter(FilterExpression filterExpression,
             Class<T> type);
 }

--- a/elide-async/src/main/java/com/yahoo/elide/async/service/dao/DefaultAsyncAPIDAO.java
+++ b/elide-async/src/main/java/com/yahoo/elide/async/service/dao/DefaultAsyncAPIDAO.java
@@ -21,7 +21,6 @@ import lombok.Setter;
 import lombok.extern.slf4j.Slf4j;
 
 import java.io.IOException;
-import java.util.Collection;
 import java.util.Collections;
 import java.util.Iterator;
 import java.util.UUID;
@@ -53,10 +52,10 @@ public class DefaultAsyncAPIDAO implements AsyncAPIDAO {
     @Override
     public <T extends AsyncAPI> T updateStatus(String asyncAPIId, QueryStatus status, Class<T> type) {
         T queryObj = (T) executeInTransaction(dataStore, (tx, scope) -> {
-            EntityProjection asyncAPICollection = EntityProjection.builder()
+            EntityProjection asyncAPIIterable = EntityProjection.builder()
                     .type(type)
                     .build();
-            T query = (T) tx.loadObject(asyncAPICollection, asyncAPIId, scope);
+            T query = (T) tx.loadObject(asyncAPIIterable, asyncAPIId, scope);
             query.setStatus(status);
             tx.save(query, scope);
             return query;
@@ -65,15 +64,15 @@ public class DefaultAsyncAPIDAO implements AsyncAPIDAO {
     }
 
     @Override
-    public <T extends AsyncAPI> Collection<T> updateStatusAsyncAPICollection(FilterExpression filterExpression,
+    public <T extends AsyncAPI> Iterable<T> updateStatusAsyncAPIByFilter(FilterExpression filterExpression,
             QueryStatus status, Class<T> type) {
-        return updateAsyncAPICollection(filterExpression, (asyncAPI) -> {
+        return updateAsyncAPIIterable(filterExpression, (asyncAPI) -> {
             asyncAPI.setStatus(status);
             }, type);
     }
 
     /**
-     * This method updates a collection of AsyncAPI objects from database and
+     * This method updates a Iterable of AsyncAPI objects from database and
      * returns the objects updated.
      * @param filterExpression Filter expression to update AsyncAPI Objects based on
      * @param updateFunction Functional interface for updating AsyncAPI Object
@@ -81,19 +80,19 @@ public class DefaultAsyncAPIDAO implements AsyncAPIDAO {
      * @return query object list updated
      */
     @SuppressWarnings("unchecked")
-    private <T extends AsyncAPI> Collection<T> updateAsyncAPICollection(FilterExpression filterExpression,
+    private <T extends AsyncAPI> Iterable<T> updateAsyncAPIIterable(FilterExpression filterExpression,
             UpdateQuery updateFunction, Class<T> type) {
-        log.debug("updateAsyncAPICollection");
+        log.debug("updateAsyncAPIIterable");
 
-        Collection<T> asyncAPIList = null;
-        asyncAPIList = (Collection<T>) executeInTransaction(dataStore,
+        Iterable<T> asyncAPIList = null;
+        asyncAPIList = (Iterable<T>) executeInTransaction(dataStore,
                 (tx, scope) -> {
-            EntityProjection asyncAPICollection = EntityProjection.builder()
+            EntityProjection asyncAPIIterable = EntityProjection.builder()
                     .type(type)
                     .filterExpression(filterExpression)
                     .build();
 
-            Iterable<Object> loaded = tx.loadObjects(asyncAPICollection, scope);
+            Iterable<Object> loaded = tx.loadObjects(asyncAPIIterable, scope);
             Iterator<Object> itr = loaded.iterator();
 
             while (itr.hasNext()) {
@@ -108,17 +107,17 @@ public class DefaultAsyncAPIDAO implements AsyncAPIDAO {
 
     @Override
     @SuppressWarnings("unchecked")
-    public <T extends AsyncAPI> Collection<T> deleteAsyncAPIAndResultCollection(
+    public <T extends AsyncAPI> Iterable<T> deleteAsyncAPIAndResultByFilter(
             FilterExpression filterExpression, Class<T> type) {
-        log.debug("deleteAsyncAPIAndResultCollection");
-        Collection<T> asyncAPIList = null;
-        asyncAPIList = (Collection<T>) executeInTransaction(dataStore, (tx, scope) -> {
-            EntityProjection asyncAPICollection = EntityProjection.builder()
+        log.debug("deleteAsyncAPIAndResultByFilter");
+        Iterable<T> asyncAPIList = null;
+        asyncAPIList = (Iterable<T>) executeInTransaction(dataStore, (tx, scope) -> {
+            EntityProjection asyncAPIIterable = EntityProjection.builder()
                     .type(type)
                     .filterExpression(filterExpression)
                     .build();
 
-            Iterable<Object> loaded = tx.loadObjects(asyncAPICollection, scope);
+            Iterable<Object> loaded = tx.loadObjects(asyncAPIIterable, scope);
             Iterator<Object> itr = loaded.iterator();
 
             while (itr.hasNext()) {
@@ -138,10 +137,10 @@ public class DefaultAsyncAPIDAO implements AsyncAPIDAO {
             String asyncAPIId, Class<T> type) {
         log.debug("updateAsyncAPIResult");
         T queryObj = (T) executeInTransaction(dataStore, (tx, scope) -> {
-            EntityProjection asyncAPICollection = EntityProjection.builder()
+            EntityProjection asyncAPIIterable = EntityProjection.builder()
                     .type(type)
                     .build();
-            T query = (T) tx.loadObject(asyncAPICollection, asyncAPIId, scope);
+            T query = (T) tx.loadObject(asyncAPIIterable, asyncAPIId, scope);
             query.setResult(asyncAPIResult);
             if (query.getStatus().equals(QueryStatus.CANCELLED)) {
                 query.setStatus(QueryStatus.CANCEL_COMPLETE);
@@ -181,18 +180,18 @@ public class DefaultAsyncAPIDAO implements AsyncAPIDAO {
 
     @Override
     @SuppressWarnings("unchecked")
-    public <T extends AsyncAPI> Collection<T> loadAsyncAPICollection(FilterExpression filterExpression,
+    public <T extends AsyncAPI> Iterable<T> loadAsyncAPIByFilter(FilterExpression filterExpression,
             Class<T> type) {
-        Collection<T> asyncAPIList = null;
-        log.debug("loadAsyncAPICollection");
+        Iterable<T> asyncAPIList = null;
+        log.debug("loadAsyncAPIByFilter");
         try {
-            asyncAPIList = (Collection<T>) executeInTransaction(dataStore, (tx, scope) -> {
+            asyncAPIList = (Iterable<T>) executeInTransaction(dataStore, (tx, scope) -> {
 
-                EntityProjection asyncAPICollection = EntityProjection.builder()
+                EntityProjection asyncAPIIterable = EntityProjection.builder()
                         .type(type)
                         .filterExpression(filterExpression)
                         .build();
-                Iterable<Object> loaded = tx.loadObjects(asyncAPICollection, scope);
+                Iterable<Object> loaded = tx.loadObjects(asyncAPIIterable, scope);
                 return loaded;
             });
         } catch (Exception e) {

--- a/elide-async/src/main/java/com/yahoo/elide/async/service/thread/AsyncAPICancelThread.java
+++ b/elide-async/src/main/java/com/yahoo/elide/async/service/thread/AsyncAPICancelThread.java
@@ -23,16 +23,15 @@ import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.extern.slf4j.Slf4j;
 
-import java.util.ArrayList;
-import java.util.Collection;
 import java.util.Collections;
 import java.util.Date;
-import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import java.util.UUID;
 import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
 import javax.ws.rs.core.MultivaluedHashMap;
 import javax.ws.rs.core.MultivaluedMap;
 
@@ -70,18 +69,15 @@ public class AsyncAPICancelThread implements Runnable {
 
             //Construct filter expression
             PathElement statusPathElement = new PathElement(type, QueryStatus.class, "status");
-            List<QueryStatus> statusList = new ArrayList<QueryStatus>();
-            statusList.add(QueryStatus.CANCELLED);
-            statusList.add(QueryStatus.PROCESSING);
-            statusList.add(QueryStatus.QUEUED);
             FilterExpression fltStatusExpression =
-                    new InPredicate(statusPathElement, statusList);
+                    new InPredicate(statusPathElement, QueryStatus.CANCELLED, QueryStatus.PROCESSING,
+                            QueryStatus.QUEUED);
 
-            Collection<T> asyncAPICollection =
-                    asyncAPIDao.loadAsyncAPICollection(fltStatusExpression, type);
+            Iterable<T> asyncAPIIterable =
+                    asyncAPIDao.loadAsyncAPIByFilter(fltStatusExpression, type);
 
             //Active AsyncAPI UUIDs
-            Set<UUID> asyncTransactionUUIDs = asyncAPICollection.stream()
+            Set<UUID> asyncTransactionUUIDs = StreamSupport.stream(asyncAPIIterable.spliterator(), false)
                     .filter(query -> query.getStatus() == QueryStatus.CANCELLED
                     || TimeUnit.SECONDS.convert(Math.abs(new Date(System.currentTimeMillis()).getTime()
                             - query.getCreatedOn().getTime()), TimeUnit.MILLISECONDS) > maxRunTimeSeconds)
@@ -94,7 +90,7 @@ public class AsyncAPICancelThread implements Runnable {
             //AsyncAPI IDs that need to be cancelled
             Set<String> queryIDsToCancel = queryUUIDsToCancel.stream()
             .map(uuid -> {
-                return asyncAPICollection.stream()
+                return StreamSupport.stream(asyncAPIIterable.spliterator(), false)
                 .filter(query -> query.getRequestId().equals(uuid.toString()))
                 .map(T::getId)
                 .findFirst().orElseThrow(IllegalStateException::new);
@@ -120,7 +116,7 @@ public class AsyncAPICancelThread implements Runnable {
                 PathElement idPathElement = new PathElement(type, String.class, "id");
                 FilterExpression fltIdExpression =
                         new InPredicate(idPathElement, queryIDsToCancel);
-                asyncAPIDao.updateStatusAsyncAPICollection(fltIdExpression, QueryStatus.CANCEL_COMPLETE,
+                asyncAPIDao.updateStatusAsyncAPIByFilter(fltIdExpression, QueryStatus.CANCEL_COMPLETE,
                         type);
             }
         } catch (Exception e) {

--- a/elide-async/src/main/java/com/yahoo/elide/async/service/thread/AsyncAPICleanerThread.java
+++ b/elide-async/src/main/java/com/yahoo/elide/async/service/thread/AsyncAPICleanerThread.java
@@ -21,10 +21,8 @@ import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.extern.slf4j.Slf4j;
 
-import java.util.ArrayList;
 import java.util.Calendar;
 import java.util.Date;
-import java.util.List;
 
 /**
  * Runnable thread for updating AsyncAPIThread status.
@@ -58,7 +56,7 @@ public class AsyncAPICleanerThread implements Runnable {
             Date cleanupDate = dateUtil.calculateFilterDate(Calendar.DATE, queryCleanupDays);
             PathElement createdOnPathElement = new PathElement(type, Long.class, "createdOn");
             FilterExpression fltDeleteExp = new LEPredicate(createdOnPathElement, cleanupDate);
-            asyncAPIDao.deleteAsyncAPIAndResultCollection(fltDeleteExp, type);
+            asyncAPIDao.deleteAsyncAPIAndResultByFilter(fltDeleteExp, type);
         } catch (Exception e) {
             log.error("Exception in scheduled cleanup: {}", e);
         }
@@ -75,13 +73,11 @@ public class AsyncAPICleanerThread implements Runnable {
             Date filterDate = dateUtil.calculateFilterDate(Calendar.MINUTE, maxRunTimeMinutes);
             PathElement createdOnPathElement = new PathElement(type, Long.class, "createdOn");
             PathElement statusPathElement = new PathElement(type, String.class, "status");
-            List<QueryStatus> statusList = new ArrayList<QueryStatus>();
-            statusList.add(QueryStatus.PROCESSING);
-            statusList.add(QueryStatus.QUEUED);
-            FilterPredicate inPredicate = new InPredicate(statusPathElement, statusList);
+            FilterPredicate inPredicate = new InPredicate(statusPathElement, QueryStatus.PROCESSING,
+                    QueryStatus.QUEUED);
             FilterPredicate lePredicate = new LEPredicate(createdOnPathElement, filterDate);
             AndFilterExpression fltTimeoutExp = new AndFilterExpression(inPredicate, lePredicate);
-            asyncAPIDao.updateStatusAsyncAPICollection(fltTimeoutExp, QueryStatus.TIMEDOUT, type);
+            asyncAPIDao.updateStatusAsyncAPIByFilter(fltTimeoutExp, QueryStatus.TIMEDOUT, type);
         } catch (Exception e) {
             log.error("Exception in scheduled cleanup: {}", e);
         }

--- a/elide-async/src/test/java/com/yahoo/elide/async/service/dao/DefaultAsyncAPIDAOTest.java
+++ b/elide-async/src/test/java/com/yahoo/elide/async/service/dao/DefaultAsyncAPIDAOTest.java
@@ -87,7 +87,7 @@ public class DefaultAsyncAPIDAOTest {
    public void testUpdateStatusAsyncQueryCollection() {
        Iterable<Object> loaded = Arrays.asList(asyncQuery, asyncQuery);
        when(tx.loadObjects(any(), any())).thenReturn(loaded);
-       asyncAPIDAO.updateStatusAsyncAPICollection(filter, QueryStatus.TIMEDOUT, asyncQuery.getClass());
+       asyncAPIDAO.updateStatusAsyncAPIByFilter(filter, QueryStatus.TIMEDOUT, asyncQuery.getClass());
        verify(tx, times(2)).save(any(AsyncQuery.class), any(RequestScope.class));
        verify(asyncQuery, times(2)).setStatus(QueryStatus.TIMEDOUT);
    }
@@ -96,7 +96,7 @@ public class DefaultAsyncAPIDAOTest {
     public void testDeleteAsyncQueryAndResultCollection() {
         Iterable<Object> loaded = Arrays.asList(asyncQuery, asyncQuery, asyncQuery);
         when(tx.loadObjects(any(), any())).thenReturn(loaded);
-        asyncAPIDAO.deleteAsyncAPIAndResultCollection(filter, asyncQuery.getClass());
+        asyncAPIDAO.deleteAsyncAPIAndResultByFilter(filter, asyncQuery.getClass());
         verify(dataStore, times(1)).beginTransaction();
         verify(tx, times(1)).loadObjects(any(), any());
         verify(tx, times(3)).delete(any(AsyncQuery.class), any(RequestScope.class));
@@ -118,7 +118,7 @@ public class DefaultAsyncAPIDAOTest {
     public void testLoadAsyncQueryCollection() {
         Iterable<Object> loaded = Arrays.asList(asyncQuery, asyncQuery, asyncQuery);
         when(tx.loadObjects(any(), any())).thenReturn(loaded);
-        asyncAPIDAO.loadAsyncAPICollection(filter, asyncQuery.getClass());
+        asyncAPIDAO.loadAsyncAPIByFilter(filter, asyncQuery.getClass());
         verify(dataStore, times(1)).beginTransaction();
         verify(tx, times(1)).loadObjects(any(), any());
     }

--- a/elide-async/src/test/java/com/yahoo/elide/async/service/thread/AsyncAPICancelThreadTest.java
+++ b/elide-async/src/test/java/com/yahoo/elide/async/service/thread/AsyncAPICancelThreadTest.java
@@ -79,12 +79,12 @@ public class AsyncAPICancelThreadTest {
         Collection<AsyncAPI> asyncCollection = new ArrayList<AsyncAPI>();
         asyncCollection.add(asyncQuery1);
         asyncCollection.add(asyncQuery2);
-        when(cancelThread.getAsyncAPIDao().loadAsyncAPICollection(any(), any())).thenReturn(asyncCollection);
+        when(cancelThread.getAsyncAPIDao().loadAsyncAPIByFilter(any(), any())).thenReturn(asyncCollection);
         cancelThread.cancelAsyncAPI(AsyncQuery.class);
         ArgumentCaptor<FilterExpression> filterCaptor = ArgumentCaptor.forClass(FilterExpression.class);
         ArgumentCaptor<QueryStatus> statusCaptor = ArgumentCaptor.forClass(QueryStatus.class);
-        verify(asyncAPIDao, times(1)).loadAsyncAPICollection(any(), any());
-        verify(asyncAPIDao, times(1)).updateStatusAsyncAPICollection(filterCaptor.capture(), statusCaptor.capture(), any());
+        verify(asyncAPIDao, times(1)).loadAsyncAPIByFilter(any(), any());
+        verify(asyncAPIDao, times(1)).updateStatusAsyncAPIByFilter(filterCaptor.capture(), statusCaptor.capture(), any());
         assertEquals("asyncQuery.id IN [[edc4a871-dff2-4054-804e-d80075cf828d]]", filterCaptor.getValue().toString());
         assertEquals("CANCEL_COMPLETE", statusCaptor.getValue().toString());
     }
@@ -105,12 +105,12 @@ public class AsyncAPICancelThreadTest {
         asyncCollection.add(asyncQuery1);
         asyncCollection.add(asyncQuery2);
         asyncCollection.add(asyncQuery3);
-        when(cancelThread.getAsyncAPIDao().loadAsyncAPICollection(any(), any())).thenReturn(asyncCollection);
+        when(cancelThread.getAsyncAPIDao().loadAsyncAPIByFilter(any(), any())).thenReturn(asyncCollection);
         cancelThread.cancelAsyncAPI(AsyncQuery.class);
         ArgumentCaptor<FilterExpression> fltStatusCaptor = ArgumentCaptor.forClass(FilterExpression.class);
-        verify(asyncAPIDao, times(1)).loadAsyncAPICollection(fltStatusCaptor.capture(), any());
-        assertEquals("asyncQuery.status IN [[CANCELLED, PROCESSING, QUEUED]]", fltStatusCaptor.getValue().toString());
-        verify(asyncAPIDao, times(1)).updateStatusAsyncAPICollection(any(), any(), any());
+        verify(asyncAPIDao, times(1)).loadAsyncAPIByFilter(fltStatusCaptor.capture(), any());
+        assertEquals("asyncQuery.status IN [CANCELLED, PROCESSING, QUEUED]", fltStatusCaptor.getValue().toString());
+        verify(asyncAPIDao, times(1)).updateStatusAsyncAPIByFilter(any(), any(), any());
 
     }
 
@@ -129,11 +129,11 @@ public class AsyncAPICancelThreadTest {
         asyncCollection.add(asyncQuery1);
         asyncCollection.add(asyncQuery2);
         asyncCollection.add(asyncQuery3);
-        when(cancelThread.getAsyncAPIDao().loadAsyncAPICollection(any(), any())).thenReturn(asyncCollection);
+        when(cancelThread.getAsyncAPIDao().loadAsyncAPIByFilter(any(), any())).thenReturn(asyncCollection);
         cancelThread.cancelAsyncAPI(AsyncQuery.class);
         ArgumentCaptor<FilterExpression> filterCaptor = ArgumentCaptor.forClass(FilterExpression.class);
         ArgumentCaptor<QueryStatus> statusCaptor = ArgumentCaptor.forClass(QueryStatus.class);
-        verify(asyncAPIDao, times(1)).updateStatusAsyncAPICollection(filterCaptor.capture(), statusCaptor.capture(), any());
+        verify(asyncAPIDao, times(1)).updateStatusAsyncAPIByFilter(filterCaptor.capture(), statusCaptor.capture(), any());
         assertEquals("asyncQuery.id IN [[edc4a871-dff2-4054-804e-d80075cf827d]]", filterCaptor.getValue().toString());
         assertEquals("CANCEL_COMPLETE", statusCaptor.getValue().toString());
 

--- a/elide-async/src/test/java/com/yahoo/elide/async/service/thread/AsyncAPICleanerThreadTest.java
+++ b/elide-async/src/test/java/com/yahoo/elide/async/service/thread/AsyncAPICleanerThreadTest.java
@@ -68,7 +68,7 @@ public class AsyncAPICleanerThreadTest {
         when(dateUtil.calculateFilterDate(Calendar.DATE, cleanerThread.getQueryCleanupDays())).thenReturn(testDate);
         ArgumentCaptor<FilterExpression> filterCaptor = ArgumentCaptor.forClass(FilterExpression.class);
         cleanerThread.deleteAsyncAPI(AsyncQuery.class);
-        verify(asyncAPIDao, times(1)).deleteAsyncAPIAndResultCollection(filterCaptor.capture(), any());
+        verify(asyncAPIDao, times(1)).deleteAsyncAPIAndResultByFilter(filterCaptor.capture(), any());
         assertEquals("asyncQuery.createdOn LE [" + testDate + "]", filterCaptor.getValue().toString());
     }
 
@@ -77,7 +77,7 @@ public class AsyncAPICleanerThreadTest {
         when(dateUtil.calculateFilterDate(Calendar.MINUTE, cleanerThread.getMaxRunTimeMinutes())).thenReturn(testDate);
         ArgumentCaptor<FilterExpression> filterCaptor = ArgumentCaptor.forClass(FilterExpression.class);
         cleanerThread.timeoutAsyncAPI(AsyncQuery.class);
-        verify(asyncAPIDao, times(1)).updateStatusAsyncAPICollection(filterCaptor.capture(), any(QueryStatus.class), any());
-        assertEquals("(asyncQuery.status IN [[PROCESSING, QUEUED]] AND asyncQuery.createdOn LE [" + testDate + "])", filterCaptor.getValue().toString());
+        verify(asyncAPIDao, times(1)).updateStatusAsyncAPIByFilter(filterCaptor.capture(), any(QueryStatus.class), any());
+        assertEquals("(asyncQuery.status IN [PROCESSING, QUEUED] AND asyncQuery.createdOn LE [" + testDate + "])", filterCaptor.getValue().toString());
     }
 }

